### PR TITLE
LibGL: Generate the API wrappers

### LINKED ISSUE #15814: Port: GLTron, on quitting crashes
Run GLTron, then close with the  X, the crash reporter pops up

Messages logged:

```
...
Couldn't open timidity.cfg
Couldn't open gravis.cfg
Couldn't open gsdrums.cfg
Couldn't open gssfx.cfg
Invalid audio device ID
Invalid audio device ID
300.557 WindowServer(13:13): ConnectionFromClient(0x00000013ec27a4f0) (id=16) misbehaved (SetWindowCursor: Bad window ID), disconnecting.
300.581 WindowServer(13:13): ConnectionFromClient(0x00000013ec27a4f0) (id=16) misbehaved (SetWindowCursor: Bad window ID), disconnecting.
300.637 gltron(60:60): SERENITY_VideoQuit
300.637 gltron(60:60): SERENITY_QuitMouse
300.637 gltron(60:60): SERENITY_DeleteDevice
300.666 [#0 gltron(60:60)]: Unrecoverable page fault, read from address V0x00000000000000fc
300.666 [#0 gltron(60:60)]: Note: Address V0x00000000000000fc looks like a possible nullptr dereference
300.668 [gltron(60:60)]: CRASH: CPU #0 Page Fault in ring 3
300.668 [#0 gltron(60:60)]: Exception code: 0004 (isr: 0000)
300.668 [#0 gltron(60:60)]:     pc=0x0023:0x00000011b3712f60 rflags=0x0000000000210246
300.668 [#0 gltron(60:60)]:  stack=0x0000000b1f7b30a8
300.668 [#0 gltron(60:60)]:    rax=0x00000011b3d0b090 rbx=0x000000067a5f8254 rcx=0x0000000b2f320070 rdx=0xfffffffffffffff0
300.668 [#0 gltron(60:60)]:    rbp=0x0000000b1f7b30c0 rsp=0x000000200cbeff30 rsi=0x00000000fffffffe rdi=0x0000000000000000
300.672 [#0 gltron(60:60)]:     r8=0x0000000f6ae30140  r9=0x000000073852d32f r10=0x00000007385363ea r11=0x0000000000200246
300.672 [#0 gltron(60:60)]:    r12=0x0000000b1f7b3108 r13=0x0000000a9e0e0be0 r14=0x0000000000000001 r15=0x0000000000000000
300.672 [#0 gltron(60:60)]:    cr0=0x0000000080010013 cr2=0x00000000000000fc cr3=0x0000000032fec000 cr4=0x00000000003406e0
300.672 [#0 gltron(60:60)]: 0x00000011b3712f60  (?)

300.676 [#0 gltron(60:60)]: Kernel + 0x0000000000c5ced0  Kernel::Process::crash(int, unsigned long, bool) +0x280
300.676 [#0 gltron(60:60)]: Kernel + 0x000000000114c96c  Kernel::handle_crash(Kernel::RegisterState const&, char const*, int, bool) +0x66c
300.676 [#0 gltron(60:60)]: Kernel + 0x000000000115b136  page_fault_handler +0xe96
300.676 [#0 gltron(60:60)]: Kernel + 0x0000000001158356  page_fault_asm_entry +0x36
300.680 [#0 Finalizer Task(5:5)]: Generating coredump for pid: 60
300.687 CrashDaemon(26:26): New coredump file: /tmp/coredump/gltron_60_1666803689
300.711 AudioServer(38:39): Audio client can't keep up!
300.735 AudioServer(38:39): Audio client can't keep up!
300.735 AudioServer(38:39): Audio client can't keep up!
300.762 AudioServer(38:39): Audio client can't keep up!
300.762 AudioServer(38:39): Audio client can't keep up!
300.782 AudioServer(38:39): Audio client can't keep up!
300.782 AudioServer(38:39): Audio client can't keep up!
300.806 AudioServer(38:39): Audio client can't keep up!
300.806 AudioServer(38:39): Audio client can't keep up!
300.822 AudioServer(38:39): Audio client can't keep up!
300.822 AudioServer(38:39): Audio client can't keep up!
300.845 AudioServer(38:39): Audio client can't keep up!
300.849 AudioServer(38:39): Audio client can't keep up!
300.865 AudioServer(38:39): Audio client can't keep up!
300.865 AudioServer(38:39): Audio client can't keep up!
300.924 AudioServer(38:39): Audio client can't keep up!
300.930 AudioServer(38:39): Audio client can't keep up!
300.930 AudioServer(38:39): Audio client can't keep up!
300.932 AudioServer(38:39): Audio client can't keep up!
300.975 AudioServer(38:39): Audio client can't keep up!
300.975 AudioServer(38:39): Audio client can't keep up!
300.991 AudioServer(38:39): Audio client can't keep up!
300.996 AudioServer(38:39): Audio client can't keep up!
301.501 CrashReporter(62:62): Started thread "", tid = 63
302.208 CrashReporter(62:63): Generating backtrace took 706 ms
302.208 CrashReporter(62:63): --- Backtrace for thread #0 (TID 60) ---
302.208 CrashReporter(62:63): 0x00000011b3712f60: [/usr/lib/libgl.so.serenity] GL::GLContext::gl_get_error() +0x0 (GLContext.cpp:176)
302.208 CrashReporter(62:63): 0x000000067a5c4043: [/usr/local/bin/gltron] drawGuiBackground +0x11 (crt0.cpp:50)
302.212 CrashReporter(62:63): 0x000000067a5c4395: [/usr/local/bin/gltron] displayGui +0x8 (crt0.cpp:50)
302.212 CrashReporter(62:63): 0x000000067a5d1d2d: [/usr/local/bin/gltron] SystemMainLoop +0x73 (nebu_SourceSample.h:21)
302.212 CrashReporter(62:63): 0x000000067a5c2abb: [/usr/local/bin/gltron] c_SystemMainLoop +0x12 (crt0.cpp:50)
302.216 CrashReporter(62:63): 0x000000067a5d2b93: [/usr/local/bin/gltron] luaD_call +0x128 (nebu_SourceSample.h:21)
302.216 CrashReporter(62:63): 0x000000067a5d776d: [/usr/local/bin/gltron] luaV_execute +0x22e (nebu_SourceSample.h:21)
302.216 CrashReporter(62:63): 0x000000067a5d2bb7: [/usr/local/bin/gltron] luaD_call +0x14c (nebu_SourceSample.h:21)
302.216 CrashReporter(62:63): 0x000000067a5d2e76: [/usr/local/bin/gltron] luaD_runprotected +0x6a (nebu_SourceSample.h:21)
302.220 CrashReporter(62:63): 0x000000067a5d311d: [/usr/local/bin/gltron] lua_call +0x38 (nebu_SourceSample.h:21)
302.220 CrashReporter(62:63): 0x000000067a5ce6ae: [/usr/local/bin/gltron] runScript +0x15 (sound_glue.cpp:294)
302.224 CrashReporter(62:63): 0x000000067a5c23a8: [/usr/local/bin/gltron] main +0x19 (instrum_dls.c:179)
302.224 CrashReporter(62:63): 0x000000067a5c2408: [/usr/local/bin/gltron] _entry +0x48 (crt0.cpp:48)

```

### PR DESCRIPTION
**Pull request by GMTA**
Generate the LibGL API wrapper methods and header file from a JSON definition. This also bumps up the supported API method count from 211 to 376 by generating a lot of variants we previously did not have.

The goal was to fix #15814, which it does by generating API wrappers that return immediately if there is no active GLContext. Unfortunately, GLTron now crashes because of a different issue when closing :^(

There is a lot of `DeprecatedString` usage in the API wrapper generator. The code was largely written before the `String/DeprecatedString` merge, and updating both `JsonObject` and `SourceGenerator` causes a yakplosion... this will need to be dealt with in another PR.

### CODE REVIEW

File: Meta/Lagom/Tools/CodeGenerators/LibGL/GenerateGLAPIWrapper.cpp [hunk 1]
```diff
@@ -0,0 +1,570 @@
...
+};
+
+struct VariantType {
+    DeprecatedString encoded_type;
+    Optional<DeprecatedString> implementation;
+    bool unimplemented;
+};
+
+struct Variants {
+    Vector<DeprecatedString> api_suffices { "" };
```
Comment (FalseHonesty) [inline Meta/Lagom/Tools/CodeGenerators/LibGL/GenerateGLAPIWrapper.cpp:43]: Should this be spelled suffixes? (And the other few places this term shows up)
```suggestion
    Vector<DeprecatedString> api_suffixes { "" };
```

Comment (GMTA) [reply Meta/Lagom/Tools/CodeGenerators/LibGL/GenerateGLAPIWrapper.cpp]: Done!

**Review by sunverwerth — APPROVED** [review]
This looks good :+1:

**Comment by sunverwerth** [comment]
Yeah I just tested this on my machine. 3DFileViewer and glquake still work and look fine :^)

**Pull request merged**